designs/helper_to_upgrade_projects_by_rescaffolding.md

```diff
@@ -0,0 +1,166 @@
+| Authors                            | Creation Date | Status      | Extra |
+|------------------------------------|---------------|-------------|---|
+| @camilamacedo86,@Kavinjsir,@varshaprasad96 | Feb, 2023     | Implementeble | - |
+
+Experimental Helper to upgrade projects by re-scaffolding
+===================
+
+This proposal amis to provide a new alpha command with a helper which
+would be able to re-scaffold the project from the scratch based on
+the [PROJECT config][project-config].
+
+## Example
+
+By running a command like following, users would be able to re-scaffold the whole project from the scratch using the
+current version of KubeBuilder binary available.
+
+```shell
+kubebuilder alpha generate [OPTIONS]
+```
+
+## Open Questions
+
+N/A
+
+## Summary
+
+Therefore, a new command can be designed to load user configs from the [PROJECT config][project-config] file, and run the corresponding kubebuilder subcommands to generate the project based on the new kubebuilder version. Thus, it makes it easier for the users to migrate their operator projects to the new scaffolding.
+
+## Motivation
+
+A common scenario is to upgrade the project based on the newer Kubebuilder. The recommended (straightforward) steps are:
+
+- a) re-scaffold all files from scratch using the upper version/plugins
+- b) copy user-defined source code to the new layout
+
+The proposed command will automate the process at maximum, therefore helping operator authors with minimizing the manual effort.
+
+The main motivation of this proposal is to provide a helper for upgrades and
+make less painful this process. Examples:
+
+- See the discussion [How to regenerate scaffolding?](https://github.com/kubernetes-sigs/kubebuilder/discussions/2864)
+- From [slack channel By Paul Laffitte](https://kubernetes.slack.com/archives/CAR30FCJZ/p1675166014762669)
+
+### Goals
+
+- Help users upgrade their project with the latest changes
+- Help users to re-scaffold the projects from the scratch based on what was done previously with the tool
+- Make less painful the process to upgrade
+
+### Non-Goals
+
+- Change the default layout or how the KubeBuilder CLI works
+- Deal with customizations or deviations from the proposed layout
+- Be able to perform the project upgrade to the latest changes without human bean interactions
```

Comment:
without human **bean** interaction~s~ -> without human **brain** interaction
I guess in this case you mean "brain" ? This word could also be just removed tho : "without human interaction"